File: tests/test_cpg/fixtures/parity_cmdi.py
"""Cross-language command injection fixture — Python."""

import os


def ping_host():
    host = sys.argv[1]  # $ source=command_injection
    os.system(f"ping -c 1 {host}")  # $ sink=command_injection


def run_admin_cmd():
    cmd = os.environ.get("ADMIN_CMD")  # $ source=command_injection
    os.popen(cmd)  # $ sink=command_injection

File: tests/test_cpg/fixtures/parity_cmdi.js
/** Cross-language command injection fixture — JavaScript (Node.js). */

var cp = require('child_process');

function pingHost() {
    var host = process.argv[2];  // $ source=command_injection
    cp.exec("ping -c 1 " + host);  // $ sink=command_injection
}

function runAdminCmd() {
    var cmd = process.env.ADMIN_CMD;  // $ source=command_injection
    cp.exec(cmd);  // $ sink=command_injection
}
